route-rs-runtime/src/link/blackhole_link.rs

```diff
@@ -0,0 +1,128 @@
+use crate::link::PacketStream;
+use futures::{Async, Future, Poll};
+
+pub struct BlackHoleLink<Packet: Sized> {
+    input_stream: PacketStream<Packet>,
+}
+
+impl<Packet: Sized> BlackHoleLink<Packet> {
+    pub fn new(input_stream: PacketStream<Packet>) -> Self {
+        BlackHoleLink { input_stream }
+    }
+}
+
+impl<Packet: Sized> Future for BlackHoleLink<Packet> {
+    type Item = ();
+    type Error = ();
+
+    /*
+    4 cases: Async::Ready(Some), Async::Ready(None), Async::NotReady, Err
+
+    Async::Ready(Some): We pull the packet and discard it, into the ether.
+    Async::Ready(None): The input_stream doesn't have anymore input. Semantically, it's like an
+    iterator has exhausted it's input. We should return "Ok(Async::Ready(None))" to signify to tokio
+    that this link is now closed.
+
+    Async::NotReady: There is more input for us to process, but we can't make any more progress right
+    now. The contract for Streams asks us to register with a Reactor so we will be woken up again by
+    an Executor, but we will be relying on Tokio to do that for us. This case is handled by the
+    "try_ready!" macro, which will automatically return "Ok(Async::NotReady)" if the input stream
+    gives us NotReady.
+
+    Err: is also handled by the "try_ready!" macro.
+    */
+    fn poll(&mut self) -> Poll<Self::Item, Self::Error> {
+        loop {
+            let input_packet_option: Option<Packet> = try_ready!(self.input_stream.poll());
+            match input_packet_option {
+                None => return Ok(Async::Ready(())),
+                Some(_) => {}
+            };
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::element::ClassifyElement;
+    use crate::link::ClassifyLink;
+    use crate::utils::test::packet_collectors::ExhaustiveCollector;
+    use crate::utils::test::packet_generators::{immediate_stream, PacketIntervalGenerator};
+    use core::time;
+    use crossbeam::crossbeam_channel;
+    use futures::future::lazy;
+
+    #[allow(dead_code)]
+    struct ClassifyEvenOddElement {
```

Comment:
This probably shouldn't be redefined every time we need to test a new link w/ classification.

Reply:
Moved to a new common file in elements.